src/NuGet.Clients/NuGet.SolutionRestoreManager/Resources.resx

```diff
@@ -124,8 +124,8 @@
     <value>Error occurred while restoring NuGet packages: {0}</value>
   </data>
   <data name="Error_PackageDownload_OnlyExactVersionsAreAllowed" xml:space="preserve">
-    <value>'{0}' is not an exact version like '[1.0.0]'. Only exact versions are allowed with PackageDownload.</value>
-    <comment>0 - the version string that's not exact</comment>
+    <value>In the package '{1}', '{0}' is not an exact version like '[1.0.0]'. Only exact versions are allowed with PackageDownload.</value>
```

Comment:
I like that you included the package ID.

It's unusual that arguments are not in ordinal order (e.g.:  `...{0}...{1}...` vs. `...{1}...{0}...`), but I won't lose sleep over it.

The wording needs improvement.  "In the package" is incorrect and potentially misleading.  I already suggested something [here](https://github.com/NuGet/NuGet.Client/pull/5351#discussion_r1291893217), but working with what you have, the simplest possible change that makes sense might be "For package '{1}', '{0}' is not..."

Maybe others have a better suggestion.